tests/unit/test_charm.py

```diff
@@ -204,6 +204,7 @@ def test_on_postgresql_pebble_ready_no_connection(harness):
 
 def test_on_get_password(harness):
     # Create a mock event and set passwords in peer relation data.
+    harness.set_leader(True)
```

Comment:
All units should be able to get the password. If now we reduce it to the leader, we're changing the expected behavior instead of the previously expected behavior of the application.

I mean: earlier this test was ensuring that all units get the password (in a way as described by the test).
Setting the unit to `leader`, from now on this test only ensures (expect) that we can get the password this way for the leader. 

Tests generally _mustn't_ change -- except when being aligned to a new behavior (i.e. fix or feature) we are implemneting. However, the changes we did in the charm should have been fully identical to the previous behavior. 

There was supposed to be no behavior change in this PR -- thus the tests have to stay intact, too.

Was there a reason for switching to leader? I may be overlooking something, perhaps we did a change that stopped the test from passing as before (and we may want to investigate that).

Reply:
Yes, this specific test was failing after the library bump.

My reason to add it was: I realized that this exact same test on the postgreSQL VM repo had `set_leader(True)` in it ([this commit,](https://github.com/canonical/postgresql-operator/commit/368dba73450d8a640bb1b1847d79454c1e639055) we can see it being added in the diff). At first I saw no reason for this test to be different on k8s and vm, so I added `set_leader(True)` here in order to make them coherent with each other, and this change made it pass again.

I thought this was a fair addition to the test, since the same exact test over on VM had it in the first place (not sure why we didn't have it here). If there was a reason for it to not be here, then I might need some more context and suggestions on what to do about it (if not having this).

I think its a good idea to have @marceloneppel and @dragomirp input here, as they might have context that I lack.

Reply:
@juditnovak I think the issue is that `fetch_my_relation_field` is marked as leader only:
https://github.com/canonical/postgresql-k8s-operator/blob/lucas/update-lib/lib/charms/data_platform_libs/v0/data_interfaces.py#L1543

Reply:
I'd propose merging this as is, until we establish the correct behaviour of the translation logic, since we already have it everywhere else.

Reply:
Sure @dragomirp if U prefer, go head.